### PR TITLE
Test `mirascope.core.base`: 100% coverages

### PR DESCRIPTION
**Pull request by willbakst**
Adds tests for all of the core base module with 100% coverage.

### CODE REVIEW

**Review by brenkao — APPROVED** [review]

File: tests/core/base/_utils/test_get_audio_type.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    assert get_audio_type(MP3_DATA) == "mp3"
+    assert get_audio_type(AIFF_DATA) == "aiff"
+    assert get_audio_type(AAC_DATA) == "aac"
+    assert get_audio_type(OGG_DATA) == "ogg"
+    assert get_audio_type(FLAC_DATA) == "flac"
+    with pytest.raises(ValueError, match="Unsupported audio type"):
+        get_audio_type(b"invalid")
+
+
+WAV_DATA = b"RIFF$\x00\x00\x00WAVEfmt \x10\x00\x00\x00\x01\x00\x01\x00D\xac\x00\x00\x88X\x01\x00\x02\x00\x10\x00data\x00\x00\x00\x00"
```
Comment (brenkao) [inline tests/core/base/_utils/test_get_audio_type.py:20]: any particular reason this is on bottom of file?

File: tests/core/base/_utils/test_get_image_type.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    assert get_image_type(PNG_DATA) == "png"
+    assert get_image_type(GIF_DATA) == "gif"
+    assert get_image_type(WEBP_DATA) == "webp"
+    assert get_image_type(HEIC_DATA) == "heic"
+    assert get_image_type(HEIF_DATA) == "heif"
+    with pytest.raises(ValueError, match="Unsupported image type"):
+        get_image_type(b"invalid")
+
+
+JPEG_DATA = b"\xff\xd8\xff\xe0\x00\x10JFIF\x00\x01\x02\x01\x00e\x00e\x00\x00\xff\xe1\x10>Exif\x00\x00"
```
Comment (brenkao) [inline tests/core/base/_utils/test_get_image_type.py:20]: same here, do we want constants to be at bottom of file?

File: tests/core/base/_utils/test_get_template_values.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+"""Tests for the get_template_values function in the _utils module."""
```
Comment (brenkao) [inline tests/core/base/_utils/test_get_template_values.py:1]: nit: """Tests the `_utils.get_template_values` module."""

File: tests/core/base/_utils/test_get_template_variables.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+"""Tests for the get_template_variables function in the _utils module."""
```
Comment (brenkao) [inline tests/core/base/_utils/test_get_template_variables.py:1]: same nit

File: tests/core/base/_utils/test_get_audio_type.py (hunk 1 above)
Comment (willbakst) [reply tests/core/base/_utils/test_get_audio_type.py:20]: Oops, I did this originally when I was using entire files, but they were way too large so I cut them down to single lines. Moved to the top of the file.

File: tests/core/base/_utils/test_get_image_type.py (hunk 2 above)
Comment (willbakst) [reply tests/core/base/_utils/test_get_image_type.py:20]: Nope, see other response, this was originally so we could see the test function when the file data was way too long

File: tests/core/base/_utils/test_get_template_values.py (hunk 3 above)
Comment (willbakst) [reply tests/core/base/_utils/test_get_template_values.py:1]: Done.

File: tests/core/base/_utils/test_get_template_variables.py (hunk 4 above)
Comment (willbakst) [reply tests/core/base/_utils/test_get_template_variables.py:1]: Done.

**Pull request merged**